go/store/nbs/archive_writer.go

```diff
@@ -79,6 +83,18 @@ When all of these steps have been completed without error, the ByteSink used to
 to complete the archive writing process.
 */
 
+// newArchiveWriter - Create an *archiveWriter with the given output ByteSync, which will be used to materialize an archive on disk.
```

Comment:
Stale comment. For the TODO, I think it's fine here to just use `NewBufferedFileByteSink("", ...)`, which will use `tempfiles.MovableTempFileProvider`, which a bunch of nbs/store is already leaning on...